### PR TITLE
Feature/bullet

### PR DESCRIPTION
**Pull request by Kristian1997**
Includes Player update, Item update and bullet update. Also has different changes to different parts such as: Timer-, Moving-, BulletMoving- and Shootingpart.

### CODE REVIEW

File: bullet/src/main/java/dk/sdu/cookie/castle/bullet/BulletProcessing.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    @Override
+    public void process(GameData gameData, World world) {
+        for (Entity entity : world.getEntities()) {
+            if (entity.getPart(ShootingPart.class) != null && entity.getPart(WeaponPart.class) != null) {
+
+                ShootingPart shootingPart = entity.getPart(ShootingPart.class);
+
+                //Shoot if isShooting is true, ie. space is pressed.
+
+                if (shootingPart.isShooting() && shootingPart.isCanShoot()) {
```
Comment (jonasolesen) [inline bullet/src/main/java/dk/sdu/cookie/castle/bullet/BulletProcessing.java]: Would change to canShoot

File: bullet/src/main/java/dk/sdu/cookie/castle/bullet/BulletProcessing.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    PositionPart positionPart = entity.getPart(PositionPart.class);
+                    Vector2f positionsVector = new Vector2f(positionPart.getRadians());
```
Comment (jonasolesen) [inline bullet/src/main/java/dk/sdu/cookie/castle/bullet/BulletProcessing.java]: Change to singular

File: bullet/src/main/java/dk/sdu/cookie/castle/bullet/BulletProcessing.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                            world.removeEntity(bullet);
+                        }
+                        break;
+                    case PLAYER_BULLET:
+                        System.out.println("Hitting another player bullet");
+                        break;
+                    default:
+                        break;
+                }
+                collisionPart.setHit(false);
```
Comment (jonasolesen) [inline bullet/src/main/java/dk/sdu/cookie/castle/bullet/BulletProcessing.java]: Consider changing to setIsHit

File: common/src/main/java/dk/sdu/cookie/castle/common/data/Entityparts/ShootingPart.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package dk.sdu.cookie.castle.common.data.Entityparts;
+
+import dk.sdu.cookie.castle.common.data.Entity;
+import dk.sdu.cookie.castle.common.data.GameData;
+
+public class ShootingPart implements EntityPart {
+
+    private boolean isShooting;
+    private boolean canShoot;
+    private float nextShoot;
```
Comment (jonasolesen) [inline common/src/main/java/dk/sdu/cookie/castle/common/data/Entityparts/ShootingPart.java]: Maybe nextShot?

File: common/src/main/java/dk/sdu/cookie/castle/common/data/Entityparts/ShootingPart.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    private float UPDATE_TIME;
+    private String ID;
+
+    public ShootingPart(String ID, float attackspeed) {
```
Comment (jonasolesen) [inline common/src/main/java/dk/sdu/cookie/castle/common/data/Entityparts/ShootingPart.java]: attackSpeed

File: item/pom.xml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <version>4.1.0</version>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Version>${project.version}</Bundle-Version>
+                        <Service-Component>OSGI-INF/plugin.xml,OSGI-INF/processing.xml</Service-Component>
```
Comment (jonasolesen) [inline item/pom.xml]: Consider 'OSGI-INF/*.xml' instead and remove unnecessary whitespace at line 28

File: item/src/main/java/dk/sdu/cookie/castle/item/ItemProcessing.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
+package dk.sdu.cookie.castle.item;
```
Comment (jonasolesen) [inline item/src/main/java/dk/sdu/cookie/castle/item/ItemProcessing.java:1]: Remove unnecessary whitespaces in this file

File: player/pom.xml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <extensions>true</extensions>
+                <version>4.1.0</version>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Version>${project.version}</Bundle-Version>
+                        <Import-Package>dk.sdu.cookie.castle.common.*,org.osgi.framework</Import-Package>
+                        <Service-Component>OSGI-INF/plugin.xml,OSGI-INF/processing.xml</Service-Component>
```
Comment (jonasolesen) [inline player/pom.xml]: Consider 'OSGI-INF/*.xml'

**Review by jonasolesen — CHANGES_REQUESTED** [review]
yes

**Review by frederikrosenberg — APPROVED** [review]

**Review by frederikrosenberg — APPROVED** [review]

**Review by jonasolesen — APPROVED** [review]
